src/Console/InfectionContainer.php

```diff
@@ -213,24 +212,33 @@ public static function create(): self
         ]);
     }
 
-    public function buildDynamicDependencies(InputInterface $input): void
-    {
-        $this['infection.config'] = static function (self $container) use ($input): InfectionConfig {
+    public function withDynamicParameters(
+        ?string $configFile,
+        ?string $mutators,
+        bool $showMutations,
+        string $logVerbosity,
+        bool $debug,
+        bool $onlyCovered,
+        string $formatter,
+        bool $noProgress,
+        string $existingCoveragePath,
+        string $initialTestsPhpOptions,
+        bool $ignoreMsiWithNoMutations,
+        float $minMsi,
+        float $minCoveredMsi
+    ): self {
+        $clone = clone $this;
```

Comment:
Is it kind of immutability for container or any other purpose?

Reply:
just for immutability